File: bin/socval/CoreAutomation.framework/Versions/A/Resources/Scripting/python_automation/coreautomation/uia_app.py
from CoreAutomationFramework import *

class UIAApp(object):
    def __init__(self, obj=None):
        self.cam_uia_app = obj
    
    def interact(self, js_text):
        return self.cam_uia_app.interact_(js_text)
        
    def __getattr__(self, name):
        def call_wrapper(*args):
            return self.cam_uia_app.function_arguments_(name, args)
        
        return call_wrapper

File: bin/socval/CoreAutomation.framework/Versions/A/Resources/Scripting/ruby_automation/cam/uia_app.rb
module CAM
  
  class UIAApp
    def initialize(app_object)
      @cam_uia_app = app_object
    end
    
    def method_missing(method, *args, &block)
      if using_macruby?
        @cam_uia_app.function(method.to_s, :arguments => args)
      else
        @cam_uia_app.function_arguments_(method.to_s, args).to_s
      end
    end
    
    def interact(js_text)
      @cam_uia_app.interact(js_text).to_s()
    end
    
  end
  
end
